File: easy/go/0067.go
func addBinary(a string, b string) string {
    var maxlen int;
    var la int = len(a)
    var lb int = len(b)
    if la > lb{
        maxlen = la
    }else{
        maxlen = lb
    }
    res := make([]byte, maxlen+1)
    var c byte
    for i:= 0; i < maxlen; i++{
        var sa, sb byte
        if i < la{
            sa = a[la - i - 1] - '0'
        }
        if i < lb{
            sb = b[lb - i - 1] - '0'
        }
        sum := sa + sb + c
        if sum >= 2{
            c = 1
        }else{
            c = 0
        }
        res[maxlen - i] = sum % 2 + '0'
    }
    if c == 1{
        res[0] = '1'
        return string(res)
    }
    return string(res[1:])
}

File: easy/python/0067.py
class Solution:
    def addBinary(self, a: 'str', b: 'str') -> 'str':
        sizeA, sizeB = len(a) - 1, len(b) - 1
        res, c = "", 0
        while sizeA > -1 or sizeB > -1:
            if sizeA < 0: tmpa = 0
            else : tmpa = int(a[sizeA])
            if sizeB < 0: tmpb = 0
            else : tmpb = int(b[sizeB])
            sum_ = tmpa + tmpb + c
            c = sum_ // 2
            sum_ = str(sum_ % 2)
            sizeA -= 1
            sizeB -= 1
            res = sum_ + res
        return "1" * c + res